src/posthog-persistence.ts

```diff
@@ -221,7 +221,11 @@ export class PostHogPersistence {
 
     update_campaign_params(): void {
         if (!this.campaign_params_saved) {
-            this.register(Info.campaignParams(this.config.custom_campaign_params))
+            const campaignParams = Info.campaignParams(this.config.custom_campaign_params)
+            // only save campaign params if there were any
+            if (!isEmptyObject(stripEmptyProperties(campaignParams))) {
+                this.register(Info.campaignParams(this.config.custom_campaign_params))
+            }
             this.campaign_params_saved = true
```

Comment:
should we only be marked them as saved when we did send something or doesn't matter?